### PR TITLE
Standardize bonus

### PR DESCRIPTION
**Pull request by Rocky640**
Small code and comment edits
Use same order bonus definitions as they appear later in the code.

No functional changes, same bench.
Fixes also a missing Weights[PawnStructure] when tracing.

### CODE REVIEW

File: src/evaluate.cpp [hunk 1]
```diff
@@ -155,20 +155,27 @@ namespace {
...
     { S( 8, 2), S(13, 4) }  // Bishops
   };
-
+  
+  // RookOnFile[semiopen/open] contains bonuses for each rook when there is no friendly
+  // pawn on the rook file.
+  const Score RookOnFile[2] = { S(19, 10), S(43, 21) };
+
+  // ThreatBySafePawn[PieceType] contains bonuses according to which piece
+  // type is attacked by a pawn which is protected or not attacked
```
Comment (mcostalba) [inline src/evaluate.cpp]: Missing end dot here.

File: src/evaluate.cpp [hunk 2]
```diff
@@ -155,20 +155,27 @@ namespace {
...
+  const Score ThreatBySafePawn[PIECE_TYPE_NB] = {
+    S(0, 0), S(0, 0), S(176, 139), S(131, 127), S(217, 218), S(203, 215) };
```
Comment (mcostalba) [inline src/evaluate.cpp]: Alignment on 2 spaces, not 3 (see other similar code)

File: src/evaluate.cpp [hunk 3]
```diff
@@ -155,20 +155,27 @@ namespace {
     { S(21, 5), S(31, 8) }, // Knights
     { S( 8, 2), S(13, 4) }  // Bishops
   };
-
+
```
Comment (mcostalba) [inline src/evaluate.cpp]: Trailing whitespace here

File: src/evaluate.cpp [hunk 4]
```diff
@@ -155,20 +155,27 @@ namespace {
...
+   { S(0, 0), S(0, 25), S(40, 62), S(40, 59), S( 0, 34), S(35, 48) }  // Rook attacks
   };
 
-  // ThreatenedByPawn[PieceType] contains a penalty according to which piece
-  // type is attacked by a pawn.
-  const Score ThreatenedByPawn[PIECE_TYPE_NB] = {
-    S(0, 0), S(0, 0), S(176, 139), S(131, 127), S(217, 218), S(203, 215)
-  };
+  // ThreatByKing[on one/on many] contains bonuses for King attacks on
+  // pawns or pieces which are not pawn defended
```
Comment (mcostalba) [inline src/evaluate.cpp]: Missing end dot here.

File: src/evaluate.cpp [hunk 5]
```diff
@@ -155,20 +155,27 @@ namespace {
...
+  const Score ThreatByKing[2] = { S(3, 62), S(9, 138) };
```
Comment (mcostalba) [inline src/evaluate.cpp]: No need [2], just [] for consistency

File: src/evaluate.cpp [hunk 6]
```diff
@@ -155,20 +155,27 @@ namespace {
...
+   { S(0, 0), S(0, 33), S(45, 43), S(46, 47), S(72,107), S(48,118) }, // Minor attacks
+   { S(0, 0), S(0, 25), S(40, 62), S(40, 59), S( 0, 34), S(35, 48) }  // Rook attacks
   };
 
-  // ThreatenedByPawn[PieceType] contains a penalty according to which piece
-  // type is attacked by a pawn.
-  const Score ThreatenedByPawn[PIECE_TYPE_NB] = {
-    S(0, 0), S(0, 0), S(176, 139), S(131, 127), S(217, 218), S(203, 215)
-  };
+  // ThreatByKing[on one/on many] contains bonuses for King attacks on
```
Comment (mcostalba) [inline src/evaluate.cpp]: Reverse descrition , should be ThreatByKing[on many / on one]

File: src/pawns.cpp [hunk 7]
```diff
@@ -45,18 +40,24 @@ namespace {
 
   // Backward pawn penalty by opposed flag
   const Score Backward[2] = { S(67, 42), S(49, 24) };
+  
+  // Unsupported pawn penalty, for pawns which are neither isolated or backward.
```
Comment (mcostalba) [inline src/pawns.cpp]: Instead here we don't need the dot :-)  On one liners no dot, on multi-line then add dot

File: src/evaluate.cpp [hunk 8]
```diff
@@ -839,7 +841,7 @@ Value Eval::evaluate(const Position& pos) {
   {
       Trace::add(MATERIAL, pos.psq_score());
       Trace::add(IMBALANCE, ei.me->imbalance());
-      Trace::add(PAWN, ei.pi->pawns_score());
+      Trace::add(PAWN, ei.pi->pawns_score() * Weights[PawnStructure]);
```
Comment (mcostalba) [inline src/evaluate.cpp]: Nice one!

**Comment by mcostalba** [comment]
Overall a nice patch, it even fixes a small bug in Tracing. Please see teh individual comments for each line

**Comment by Rocky640** [comment]
Thanks for your expert review.
The comment dots and the whitespaces had been fixed.

But I'm not sure about the other
a) "Alignment on 2 spaces, not 3 (see other similar code)" ?
    Do you mean (#,###) instead of (#, ###) ?
b) "No need [2], just [] for consistency" 
     (because there are many other one dimensional arrays declaration which ends with [2] or [something])
c) ThreatByKing description is ok.

**Comment by mcostalba** [comment]
Yes, you are right. Patch is fine now.

Joona, I suggest to commit in SF 7 due to a bug fix in tracing pawns scores:

   - Trace::add(PAWN, ei.pi->pawns_score());

   + Trace::add(PAWN, ei.pi->pawns_score() * Weights[PawnStructure]);

**Comment by zamar** [comment]
Thanks @Rocky640, @mcostalba: Thanks. It's a non-functional change (apart from tracing fixes), so we can include this for beta2.

**Comment by Rocky640** [comment]
@mcostalba  please review changes pushed just above. I'm not sure about the proper score formatting for Lever and Outpost.

**Comment by braich** [comment]
https://github.com/official-stockfish/Stockfish/blob/master/src/benchmark.cpp#L78
The comma isn't needed here.

**Comment by braich** [comment]
https://github.com/official-stockfish/Stockfish/blob/master/src/endgame.cpp#L42
Here it is not needed too.

**Comment by braich** [comment]
It seems that your pull request is about dots and commas :smile:. Could you add these commas to your pull request?

**Comment by Rocky640** [comment]
Sure, thank you for your contribution !

**Pull request closed without merging**